### PR TITLE
Fixed auto-routing when user supplies some routes

### PR DESCRIPTION
**Pull request by JoeB989**
Fixes issue #6882.
And fixed tiny intra-tile route (issue #6890)

### CODE REVIEW

File: lib/engine/auto_router.rb [hunk 1]
```diff
@@ -34,21 +34,26 @@ def compute(corporation, **opts)
...
       # which can cause all train routes to conflict with each other
       node_timeout = path_timeout / nodes.size
+      path_walk_timed_out = false
 
       now = Time.now
 
       skip_paths = static.flat_map(&:paths).to_h { |path| [path, true] }
+      # if only routing for subset of trains, keep track of which ones we'll assemble routes for
+      skip_trains = static.flat_map(:train).to_a
+      look_trains = trains - skip_trains
```
Comment (tobymao) [inline lib/engine/auto_router.rb:44]: can you jsut do trains = trains - skip_trains

File: lib/engine/auto_router.rb [hunk 2]
```diff
@@ -155,13 +161,19 @@ def compute(corporation, **opts)
         puts ' Node timeout reached' if node_abort
       end
 
+      @flash&.call('Auto route path walk failed to complete (PATH TIMEOUT)') if path_walk_timed_out
```
Comment (tobymao) [inline lib/engine/auto_router.rb:164]: does this actually work given that the calls are blocking?

File: lib/engine/auto_router.rb (hunk 1 above)
Comment (JoeB989) [reply lib/engine/auto_router.rb:44]: Sure

File: lib/engine/auto_router.rb (hunk 2 above)
Comment (JoeB989) [reply lib/engine/auto_router.rb:164]: It displays, but the combo loop is normally so fast I can't tell if the flash blocks until function ends.

Comment (tobymao) [reply lib/engine/auto_router.rb:164]: i'd put it at the end anyways

Comment (JoeB989) [reply lib/engine/auto_router.rb:164]: I just verified that it won't flash until the function ends.  If both timeouts occur, I'm not sure if user will see both flashes.

Comment (tobymao) [reply lib/engine/auto_router.rb:164]: just combine them into one

Comment (JoeB989) [reply lib/engine/auto_router.rb:164]: ok

File: lib/engine/auto_router.rb (hunk 1 above)
Comment (JoeB989) [reply lib/engine/auto_router.rb:44]: Actually it turns out I can't, there's some weird crossover side effect afterwards when I click a train to clear it (an exception where '5' is not a train).  It's like `trains = trains - skip_trains` is modifying the actual train collection in the RR company.

**Review by tobymao — APPROVED** [review]

**Pull request merged**